src/pkg/mcp/tools/deploy.go

```diff
@@ -57,14 +57,14 @@ func setupDeployTool(s *server.MCPServer, cluster string) {
 
 		client, err := cli.Connect(ctx, cluster)
 		if err != nil {
-			return mcp.NewToolResultErrorFromErr("Could not connect", err), nil
+			return mcp.NewToolResultErrorFromErr("Could not connect", err), err
```

Comment:
not sure if you're supposed to have both a tool-result and an error. In Go, usually the caller only checks one or the other, not both.